Drivers/Src/Pedals.c

```diff
@@ -20,14 +33,22 @@ void Pedals_Init(){
 /**
  * @brief   Fetches the millivoltage value of the potentiomenter as provided 
  *          by the ADC channel of the requested pedal (Accelerator or Brake),
- *          converts it to a percentage of the total distance pressed and 
- *          returns it
+ *          converts it to a percentage of the total distance pressed using 
+ *          data from calibration testing, and returns it
  * @param   pedal_t, ACCELERATOR or BRAKE as defined in enum
  * @return  distance the pedal has been pressed in percentage
  */
 int8_t Pedals_Read(pedal_t pedal){
+    if (pedal >= NUMBER_OF_PEDALS) return 0;
     int16_t millivoltsPedal = BSP_ADC_Get_Millivoltage(pedal);
-    int8_t percentage = millivoltsPedal * 100 / ADC_RANGE_MILLIVOLTS;
+
+    int8_t percentage = 0;
+    
+    if (millivoltsPedal >= LowerBound[pedal]) {
+        percentage = (millivoltsPedal - LowerBound[pedal]) * 100 /
```

Comment:
If `millivoltsPedal` is a lot higher than `LowerBound[pedal]` (eg. `ADC_RANGE_MILLIVOLTS`), I think you will get overflow when you multiply by 100.

Reply:
millivoltsPedal is going to be a value between 0 and 3300, so multiplying it by 100 will easily still be within the range for an integer. The final result should always be in the range of 0-100, fitting within an int8_t

Reply:
You are doing this with 16 bit signed integers though (max around 32,000), not 32 bit integers

Reply:
I was under the impression that integers in C get promoted by default during these kinds of expressions (the literal 100 is an int, and so everything else becomes an int), but I can add some casts to make it more explicit.